library/systemd/src/modules/systemd_service.rb

```diff
@@ -62,32 +65,64 @@ module Yast
   #
   #    service = Yast::SystemdService.find('sshd', :type=>'Type')
   #    service.properties.type  # 'simple'
-  #
-  ##
-
-  class SystemdServiceNotFound < StandardError
-    def initialize(service_name)
-      super "Service unit '#{service_name}' not found"
-    end
-  end
-
   class SystemdServiceClass < Module
+    include Yast::Logger
+
     UNIT_SUFFIX = ".service".freeze
 
-    def find(service_name, properties = {})
+    # @param service_name [String] "foo" or "foo.service"
+    # @param propmap [SystemdUnit::PropMap]
+    # @return [Service,nil]
+    def find(service_name, propmap = {})
       service_name += UNIT_SUFFIX unless service_name.end_with?(UNIT_SUFFIX)
-      service = Service.new(service_name, properties)
+      service = Service.new(service_name, propmap)
       return nil if service.properties.not_found?
       service
     end
 
-    def find!(service_name, properties = {})
-      find(service_name, properties) || raise(SystemdServiceNotFound, service_name)
+    # @param service_name [String] "foo" or "foo.service"
+    # @param propmap [SystemdUnit::PropMap]
+    # @return [Service]
+    # @raise [SystemdServiceNotFound]
+    def find!(service_name, propmap = {})
+      find(service_name, propmap) || raise(SystemdServiceNotFound, service_name)
+    end
+
+    # @param service_names [Array<String>] "foo" or "foo.service"
+    # @param propmap [SystemdUnit::PropMap]
+    # @return [Array<Service>]
+    # @raise [SystemdServiceNotFound] if an unexpected problem occurs
+    def find_many!(service_names, propmap = {})
+      snames = service_names.map { |n| n + UNIT_SUFFIX unless n.end_with?(UNIT_SUFFIX) }
+      snames_s = snames.join(" ")
+      pnames_s = SystemdUnit::DEFAULT_PROPMAP.merge(propmap).values.join(",")
+      out = Systemctl.execute("show  --property=#{pnames_s} #{snames_s}")
+      log.error "returned #{out.exit}, #{out.stderr}" unless out.exit.zero? && out.stderr.empty?
+      property_texts = out.stdout.split("\n\n")
+      raise(SystemdServiceNotFound, "?") unless snames.size == property_texts.size
+
+      snames.zip(property_texts).map do |service_name, property_text|
+        service = Service.new(service_name, propmap, property_text)
+        next nil if service.properties.not_found?
+        service
+      end
+    end
+
+    # @param service_names [Array<String>] "foo" or "foo.service"
+    # @param propmap [SystemdUnit::PropMap]
+    # @return [Array<Service,nil>]
+    def find_many(service_names, propmap = {})
+      find_many!(service_names, propmap)
+    rescue SystemdServiceNotFound
```

Comment:
to be honest I prefer other way around like in find, where it tries without exceptions and only if it is ! variant and nil is there, it raise exception.

Reply:
1. From the outside it is consistent: with the `!` it may `raise`, otherwise not
2. It is a subtle point that I need to add a comment about: we do not know WHICH service was not found. One of them made `systemctl show` return incomplete data but it is hard to match correctly.

Reply:
But you're right, it is confusing, and I don't need an exception for this. Used a condition instead.